libr/include/r_util/r_annotated_code.h

```diff
@@ -0,0 +1,123 @@
+
+#ifndef R2GHIDRA_ANNOTATEDCODE_H
+#define R2GHIDRA_ANNOTATEDCODE_H
+
+// #include <r_core.h>
+#include <r_types.h>
+#include <r_vector.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum r_syntax_highlight_type_t {
+	R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD,
+	R_SYNTAX_HIGHLIGHT_TYPE_COMMENT,
+	R_SYNTAX_HIGHLIGHT_TYPE_DATATYPE,
+	R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_NAME,
+	R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_PARAMETER,
+	R_SYNTAX_HIGHLIGHT_TYPE_LOCAL_VARIABLE,
+	R_SYNTAX_HIGHLIGHT_TYPE_CONSTANT_VARIABLE,
+	R_SYNTAX_HIGHLIGHT_TYPE_GLOBAL_VARIABLE,
+} RSyntaxHighlightType;
+
+typedef enum r_code_annotation_type_t {
+	R_CODE_ANNOTATION_TYPE_OFFSET,
+	R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT,
+	// ...
+} RCodeAnnotationType;
+
+typedef struct r_code_annotation_t {
+	size_t start;
+	size_t end;
+	RCodeAnnotationType type;
+	union {
+		struct {
+			ut64 offset;
+		} offset;
+
+		struct {
+			RSyntaxHighlightType type;
+		} syntax_highlight;
+	};
+} RCodeAnnotation;
+
+typedef struct r_annotated_code_t {
+	char *code; // owned
+	RVector /*<RCodeAnnotation>*/ annotations;
+} RAnnotatedCode;
+
+/**
+* r_annotated_code_new() - Creates a new RAnnotatedCode structure and returns it's pointer.
+* @code: Literal code for which the RAnnotatedCode structure will be created .
+*
+* This functions creates a new RAnnotatedCode structure.
+* RAnnotatedCode.code will be initialized as the character array passed.
+* This will initialize RVector /*<RCodeAnnotation>*\/ annotations as well.
+* 
+* Return: Pointer to the new RAnnotatedCode structure created.
+*/
+R_API RAnnotatedCode *r_annotated_code_new(char *code);
```

Comment:
I think it's important to note in the doc above that `code` will be owned and freed by RAnnotatedCode, so it must be a free-able string.

Reply:
Done.